WordPress/Classes/ViewRelated/Jetpack/Branding/Fullscreen Overlay/JetpackFullscreenOverlayGeneralViewModel+Analytics.swift

```diff
@@ -26,24 +26,39 @@ extension JetpackFullscreenOverlayGeneralViewModel {
 
     // MARK: Analytics Implementation
 
-    func trackOverlayDisplayed() {
+    func didDisplayOverlay() {
         WPAnalytics.track(.jetpackFullscreenOverlayDisplayed, properties: defaultProperties)
     }
 
-    func trackLearnMoreTapped() {
+    func didTapLink() {
+        guard let urlString = learnMoreButtonURL,
+              let url = URL(string: urlString) else {
+            return
+        }
+
+        let source = "jetpack_overlay_\(analyticsSource)"
+        coordinator?.navigateToLinkRoute(url: url, source: source)
         WPAnalytics.track(.jetpackFullscreenOverlayLinkTapped, properties: defaultProperties)
     }
 
-    func trackSwitchButtonTapped() {
-        WPAnalytics.track(.jetpackFullscreenOverlayButtonTapped, properties: defaultProperties)
+    func didTapPrimary() {
+        // Try to export WordPress data to a shared location before redirecting the user.
```

Comment:
How about using `coordinator.navigateToPrimaryRoute()`?

Reply:
Hi @hassaanelgarem, apologies for missing your comments here. I can open a separate PR for this targeting 21.9 if that's fine with you.